src/jit/gentree.h

```diff
@@ -1747,6 +1753,15 @@ struct GenTree
     // sets "*pIsEntire" to true if this assignment writes the full width of the local.
     bool DefinesLocalAddr(Compiler* comp, unsigned width, GenTreeLclVarCommon** pLclVarTree, bool* pIsEntire);
 
+    regNumber RegAtDef() const
+    {
+        return GetRawRegNum();
+    }
+    regNumber RegAtUse() const
+    {
+        return isContained() ? REG_NA : GetRawRegNum();
+    }
+
```

Comment:
Note that isContainedSpillTemp() check is cheap that it just needs to check whether two flags are set on a tree node.  If we still want to avoid that check we can add debug asserts that gtRegNum is not queried on a spill temp.

`assert(!isContainedSpillTemp())`

The expectation is that codegen will use logic as given below to avoid querying gtRegNum on its reg-optional operands

`op1Reg = op1->isContained() : REG_NA : op1->gtRegNum.`

Assert in GetRegNum() will catch cases where codegen is not doing above.